biomage/rds/login.py

```diff
@@ -0,0 +1,89 @@
+import sys
+import signal
+import boto3
+import click
+
+import threading
+
+from ..utils.constants import STAGING
+
+from subprocess import PIPE, run
+
+@click.command()
+
+@click.option(
+    "-i",
+    "--input_env",
+    required=False,
+    default=STAGING,
+    show_default=True,
+    help="Input environment of the RDS server.",
+)
+
+@click.option(
+    "-p",
+    "--port",
+    required=False,
+    default=5432,
+    show_default=True,
+    help="Port of the db.",
+)
+
+@click.option(
+    "-u",
+    "--user",
+    required=False,
+    default="dev_role",
+    show_default=True,
+    help="User to connect as (role is the same as user).",
+)
+
+@click.option(
+    "-r",
+    "--region",
+    required=False,
+    default="eu-west-1",
+    show_default=True,
+    help="Role to connect as (role is the same as user).",
+)
+
+def login(input_env, port, user, region):
+    """
+    Logs into a database using psql and IAM if necessary.\n
+
+    E.g.:
+    biomage rds login
+    """
+    password = None
+
+    internal_port = port
+
+    if input_env == "development":
+        password = "password"
+    else:
+        internal_port = 5432
+        print("Only local port 5432 works connecting to staging and prod for now, so setting it to 5432")
+
+        rds_client = boto3.client("rds")
+
+        remote_endpoint = get_rds_writer_endpoint(input_env, rds_client)
+
+        print(f"Generating temporary token for {input_env}")
+        password = rds_client.generate_db_auth_token(remote_endpoint, internal_port, user, region)
+
+    print("Token generated")
+
+    run(f"PGPASSWORD=\"{password}\" psql --host=localhost --port={internal_port} --username={user} --dbname=aurora_db", shell=True)
+
+def get_rds_writer_endpoint(input_env, rds_client):
```

Comment:
I would suggest using reader by default

Reply:
it turns out that by default the, when we only have 1 instance in RDS, the reader points to the write instance so this won't work as a safety measure. We'll have to restrict permissions using a `reader_role` or something like that in the future

Reply:
yep

Reply:
I think it makes more sense to have one single method to restrict read/write permissions anyways